pkg/cloud/services/network/natgateways_test.go

```diff
@@ -40,6 +40,161 @@ const (
 	ElasticIPAllocationID = "elastic-ip-allocation-id"
 )
 
+func TestFindNatGatewayForEdgeSubnet(t *testing.T) {
+	knownSubnets := []infrav1.SubnetSpec{
+		{
+			ID:               "subnet-1",
+			AvailabilityZone: "us-east-1a",
+			CidrBlock:        "10.0.10.0/24",
+			IsPublic:         false,
+			NatGatewayID:     aws.String("nat-1234567890abcdef0"),
+		},
+		{
+			ID:               "subnet-2",
+			AvailabilityZone: "us-east-1b",
+			CidrBlock:        "10.0.20.0/24",
+			IsPublic:         false,
+		},
+		{
+			ID:               "subnet-3",
+			AvailabilityZone: "us-east-1c",
+			CidrBlock:        "10.0.30.0/24",
+			IsPublic:         true,
+			NatGatewayID:     aws.String("nat-1234567890abcdef1"),
+		},
+		{
+			ID:               "subnet-4",
+			AvailabilityZone: "us-east-1d",
+			CidrBlock:        "10.0.40.0/24",
+			IsPublic:         true,
+			NatGatewayID:     aws.String("nat-1234567890abcdef2"),
+			ParentZoneName:   aws.String("zone-subnet-4"),
```

Comment:
As we are using valid zone names, let's use it to the ParentZoneName too
```suggestion
			ParentZoneName:   aws.String("us-east-1"),
```